### PR TITLE
Added the doc guideline

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added the doc guideline

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Closes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/3

### CODE REVIEW

**Comment by SAY-droid427** [comment]
@HarshCasper I have added a basic doc guideline as of now. Please check and tell me if I need to add anything else

**Comment by TidbitsJS** [comment]
Hey, @SAY-droid427, I have created a rough sketch for the article layout.
You can check it here - [Repository](https://github.com/TidbitsJS/NeoAlgo-Docs.git)
A mixed work of both of us can yield a beautiful guideline.
Do take a look :)

**Comment by SAY-droid427** [comment]
@TidbitsJS I have something like this in mind. I am waiting for @HarshCasper 's next instructions, so that I can start working on that. As I have mentioned it is just a basic guideline doc and contains the basic instructions only as of now. In my later pull requests, I will specify the article layouts as well.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SAY-droid427 Format this using Prettier.

File: docs/doc-guideline.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+   4. Function and variable names should follow camelcasing. Example: uglyNumbers()
+
+   5. Indentation should be done using 4 spaces.
+
+   6. In case of the code being written in multiple languages, the output of all the codes should be the same
+
+   7. There should not be any spaces for function call or function declaration
+
+   8. Avoid the use of typedef
```
Comment (iamrajiv) [inline docs/doc-guideline.md:34]: @SAY-droid427 Use backtick for special keywords like this `typedef`, `cout` etc.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: docs/doc-guideline.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# DOC GUIDELINE
+
+## How to write an article for the website
+
+1. All the problems written should necessarily include 3 sections
+   1. Explanation of the article
+   2. Sample input
+   3. Sample output
```
Comment (HarshCasper) [inline docs/doc-guideline.md:8]: I guess we can make it six sections: 

- Introduction about the Algorithm/Data Structure
- Explanation about the Algorithm/Data Structure
- Algorithmic Steps for implementation
- Code (C++/Java and Python as of now) 
- Input-Output Samples (Atleast two) 
- Time & Space Complexity 

This is my rough idea and a lot of people are following this. Make them descriptive and presentable

File: docs/doc-guideline.md (hunk 1 above)
Comment (HarshCasper) [inline docs/doc-guideline.md:34]: Regarding Code Guidelines have a look at some docs I wrote here: https://docs.google.com/document/d/1WYYc2ymImxS6-TKB6pRX77A9zpXb_ujGjdpfMaTDXwc/edit?usp=sharing

Go over Page 8 and you will see my doc on Code Formatting. You can copy paste stuff here and add more information as per your understanding and observation as well. 

Remember that we are not working on NeoAlgo here, but on Docs. So see things from the perspective of a Content Strategist 😉

File: docs/doc-guideline.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# Reviewing Process
+
+1. Submit the article for review after writing it.
+
+2. The team will review your article and give you feedback for the article
+
+3. Work upon the feedback and improve your article. Submit it again for further review.
+
+4. If all the required improvements have been made and the team approves your article, then your article will be published.
```
Comment (HarshCasper) [inline docs/doc-guideline.md:52]: Can you add some steps about the CI pipeline as well? We use Vercel for deployment and will add some GH actions. So just mention that the checks should pass and they can make their PR as a `draft` unless all the checks pass. If its failing and they cannot debug it, let us know

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Overall good work! I found it great that you are helping us with the content strategy 💯 

So basically the Doc Guideline should cover: 

- How to structure articles? 
- How to add Code to the articles? 
- How to add images and diagrams to the articles? 
- Using Grammarly and Language Tool for good content
- Asking for review 

The Doc Guideline is the first among the three guidelines we need: The other two being the `Style Guideline` and the `Contribution Guideline`. Let me know what you think and perhaps @TidbitsJS can coordinate with you on the efforts (Both of you will be equally compensated for the same 😉 ) 

On a side not, don't keep this file in the `docs/`. Keep it on the root directory so that everyone can see it while starting up.

File: docs/doc-guideline.md (hunk 2 above)
Comment (SAY-droid427) [reply docs/doc-guideline.md:8]: This sounds elaborate. I will make the changes

File: docs/doc-guideline.md (hunk 3 above)
Comment (SAY-droid427) [reply docs/doc-guideline.md:52]: Yes I will do so and push my changes by tomorrow

File: docs/doc-guideline.md (hunk 1 above)
Comment (SAY-droid427) [reply docs/doc-guideline.md:34]: Will do so

**Comment by SAY-droid427** [comment]
@HarshCasper  I have made the changes you had asked. Could you please review it ?

**Review by iamrajiv — APPROVED** [review]

**Comment by SAY-droid427** [comment]
@HarshCasper Please review this PR

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**